Doc/library/sqlite3.rst

```diff
@@ -2265,13 +2339,74 @@ can be found in the `SQLite URI documentation`_.
 Explanation
 -----------
 
+.. _sqlite3-transaction-control:
 .. _sqlite3-controlling-transactions:
 
 Transaction control
 ^^^^^^^^^^^^^^^^^^^
 
-The :mod:`!sqlite3` module does not adhere to the transaction handling recommended
-by :pep:`249`.
+:mod:`!sqlite3` offers multiple methods of controlling whether,
+when and how database transactions are opened and closed.
+:ref:`sqlite3-transaction-control-autocommit` is recommended,
+while :ref:`sqlite3-transaction-control-isolation-level`
+retains the pre-Python 3.12 behaviour.
+
+.. _sqlite3-transaction-control-autocommit:
+
+Transaction control via the ``autocommit`` attribute
+""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+The recommended way of controlling transaction behaviour is through
+the :attr:`Connection.autocommit` attribute,
+which should preferrably be set using the *autocommit* parameter
+of :func:`connect`.
+
+It is suggested to set *autocommit* to ``False``,
+which implies :pep:`249`-compliant transaction control.
+This means:
+
+* :mod:`!sqlite3` ensures that a transaction is always open,
+  so :meth:`Connection.commit` and :meth:`Connection.rollback`
+  will implicitly open a new transaction immediately after closing
+  the pending one.
+  :mod:`!sqlite3` uses ``BEGIN DEFERRED`` statements when opening transactions.
+* Transactions should be committed explicitly using :meth:`!commit`.
+* Transactions should be rolled back explicitly using :meth:`!rollback`.
+* An implicit rollback is performed if the database is
+  :meth:`~Connection.close`-ed with pending changes.
+
+Set *autocommit* to ``True`` to enable SQLite's `autocommit mode`_.
+In this mode, :meth:`Connection.commit` and :meth:`Connection.rollback`
+have no effect.
+
+Set *autocommit* to :data:`LEGACY_TRANSACTION_CONTROL`
+to leave transaction control behaviour to the
+:attr:`Connection.isolation_level` attribute.
+See :ref:`sqlite3-transaction-control-isolation-level` for more information.
+
+.. note::
+   The :pep:`249`-compliant :attr:`!autocommit` attribute and the
+   SQLite `autocommit mode`_ are two related but distinct concepts.
+   Use :attr:`in_transaction` to query the low-level SQLite
+   autocommit mode.
```

Comment:
One last nag—putting the note here leaves the next subhead bracketed by admonitions both top and bottom, which IMO makes the document structure harder to easily follow and is visually uncomfortable. Wouldn't it be better to put it one paragraph up, directly below the paragraph it is most relevant to (where the SQLite autocommit mode is introduced), rather than separating the two by a much less related paragraph?

Alternatively, if there is still a reason to separate them, we should link `` `autocommit mode`_ ``.

Reply:
Hm, how about making it a paragraph of the text instead of a note? I'm not sure it deserves being promoted visually as a note.

Reply:
Hmm, that could work if you don't feel it deserves to be called out as a note. However, in that case the text needs to be moved to the previous autocommit=True paragraph and rewritten as integrated prose, since that is where it is actually relevant to mention inline to make that distinction, given that is where both potentially-confused terms are discussed, since it is no longer a standalone callout.

I've made a suggestion to that effect, as well as fixed a broken reference to the attribute, due to no longer being inside its class.